addon-resizer/nanny/kubernetes_client.go

```diff
@@ -152,15 +154,65 @@ func (k *kubernetesClient) UpdateDeployment(resources *corev1.ResourceRequiremen
 	for i, container := range dep.Spec.Template.Spec.Containers {
 		if container.Name == k.container {
 			// Update the deployment.
-			dep.Spec.Template.Spec.Containers[i].Resources = *resources
-			_, err = k.clientset.AppsV1().Deployments(k.namespace).Update(dep)
-			return err
+			patches := getContainerResourcesPatches(k.container, i, mergeResources(&container.Resources, resources))
+			return k.patchDeployment(patches)
 		}
 	}
 
 	return fmt.Errorf("Container %s was not found in the deployment %s in namespace %s.", k.container, k.deployment, k.namespace)
 }
 
+func (k *kubernetesClient) patchDeployment(patches []patchRecord) error {
+	bytes, err := json.Marshal(patches)
+	if err != nil {
+		return fmt.Errorf("Cannot marshal deployment patches %+v. Reason: %+v", patches, err)
+	}
+
+	_, err = k.clientset.AppsV1().Deployments(k.namespace).Patch(k.deployment, types.JSONPatchType, bytes)
+	return err
+}
+
+type patchRecord struct {
+	Op    string      `json:"op,inline"`
+	Path  string      `json:"path,inline"`
```

Comment:
what is the `inline` bit for?

Reply:
I agree we should remove it (for both fields).

Reply:
Done